hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -68,7 +79,16 @@ cardanoLedger globals ledgerEnv =
       Right (Ledger.LedgerState{Ledger.lsUTxOState = us}, _validatedTx) ->
         Right . fromLedgerUTxO $ Ledger.utxosUtxo us
    where
-    toValidationError = ValidationError . show
+    -- As we use applyTx we only expect one ledger rule to run and one tx to
+    -- fail validation, hence using the heads of non empty lists is fine.
+    toValidationError (Ledger.ApplyTxError (e :| _)) = case e of
+      (ConwayUtxowFailure (UtxoFailure (UtxosFailure (ValidationTagMismatch _ (FailedUnexpectedly (PlutusFailure msg ctx :| _)))))) ->
+        ValidationError $
+          "Plutus validation failed: "
+            <> msg
+            <> "Debug info: "
+            <> show (debugPlutus @StandardCrypto (decodeUtf8 ctx))
```

Comment:
`debugPlutus` returns IO action, perhaps things have changed in the meantime and the signature changed and now it seems we need `unsafePerformIO` or similar to display plutus falilure.

Reply:
Yeah... I might ask them by sending a PR.

Reply:
Asked on discord: https://discord.com/channels/1136727663583698984/1239889030984896614/1330985414680510556